polkadot/xcm/pallet-xcm/src/lib.rs

```diff
@@ -3105,6 +3126,16 @@ impl<T: Config> CheckSuspension for Pallet<T> {
 	}
 }
 
+impl<T: Config> xcm_executor::traits::RecordXcm for Pallet<T> {
+	fn should_record() -> bool {
+		ShouldRecordXcm::<T>::get()
+	}
+
+	fn record(xcm: Xcm<()>) {
+		RecordedXcm::<T>::put(xcm);
+	}
+}
```

Comment:
Might be worth for the trait to expose the "setters" and the "getters".

```suggestion
impl<T: Config> xcm_executor::traits::RecordXcm for Pallet<T> {
	fn should_record() -> bool {
		ShouldRecordXcm::<T>::get()
	}

	fn set_record_xcm(enabled: bool) {
		ShouldRecordXcm::<T>::put(enabled);
	}

	fn recorded_xcm() -> Xcm<()> {
		RecordedXcm::<T>::get()
	}

	fn record(xcm: Xcm<()>) {
		RecordedXcm::<T>::put(xcm);
	}
}
```